core/storage/wal.rs

```diff
@@ -1107,6 +1107,12 @@ impl Wal for WalFile {
         tracing::debug!("read_frame({})", frame_id);
         let offset = self.frame_offset(frame_id);
         let (frame_ptr, frame_len) = (frame.as_mut_ptr(), frame.len());
+
+        let encryption_ctx = {
+            let io_ctx = self.io_ctx.borrow();
+            io_ctx.encryption_context().cloned()
+        };
+        let (header, _) = sqlite3_ondisk::parse_wal_frame_header(frame);
```

Comment:
this is likely incorrect as `frame` is just allocated memory but it is not loaded yet. That happens in the completion callback